### ISSUE TITLE
Void Linux + Wayland: Audio Output does not switch with panel button or hotkey

### ISSUE THREAD

**Issue by todfox-net** [Blackstareye/toggleAudio-blackeyeprojects.de#11]
The extension does not seem to switch between audio outputs.

I have my headphone and speaker outputs selected in the extension, have the indicators on. In playing a YT video as a test, I've tried toggling the panel button, and the audio does not change. I've also tried the hotkey (Ctrl+Alt+T, which is enabled) and the audio output and indicator in the top bar don't change, either.

GNOME 46 (Wayland) on Void Linux

**Comment by Blackstareye** [comment]
That sounds rather odd. 

What kind of devices do you use?

can you give me a neofetch output  of your system ?

When I got time I will set up a VM and try GNOME 46 (Wayland) on Void Linux.

Never heard of void linux though. Maybe this  specific os can be the problem as well :/

**Comment by todfox-net** [comment]
I use motherboard audio for the speaker and a Razer headset.

OS: Void Linux x86_64 
Host: B550M AORUS ELITE -CF 
Kernel: 6.10.6_1 
Uptime: 1 day, 11 hours, 33 mins 
Packages: 1297 (xbps-query), 17 (flatpak) 
Shell: bash 5.2.21 
Resolution: 1920x1080 
DE: GNOME 46.2 
WM: Mutter 
WM Theme: Adwaita 
Theme: Adwaita [GTK2/3] 
Icons: Adwaita [GTK2/3] 
Terminal: kgx 
CPU: AMD Ryzen 5 5600G with Radeon Graphics (12) @ 4.464GHz 
GPU: AMD ATI Radeon Vega Series / Radeon Vega Mobile Series 
GPU: NVIDIA GeForce GTX 1660 SUPER 
Memory: 3106MiB / 15357MiB 
                                                                 
                                                                        
06:00.6 Audio device [0403]: Advanced Micro Devices, Inc. [AMD] Family 17h/19h HD Audio Controller [1022:15e3]
	DeviceName: Realtek ALC1220
	Subsystem: Gigabyte Technology Co., Ltd Device [1458:a194]
	Kernel driver in use: snd_hda_intel
	Kernel modules: snd_hda_intel

Bus 003 Device 002: ID 1532:052c Razer USA, Ltd Razer Kraken V3 Pro
Device Descriptor:
  bLength                18
  bDescriptorType         1
  bcdUSB               2.00
  bDeviceClass          239 Miscellaneous Device
  bDeviceSubClass         2 [unknown]
  bDeviceProtocol         1 Interface Association
  bMaxPacketSize0        64
  idVendor           0x1532 Razer USA, Ltd
  idProduct          0x052c Razer Kraken V3 Pro
  bcdDevice            9.00
  iManufacturer           1 NXP SEMICONDUCTORS
  iProduct                2 Razer Kraken V3 Pro

**Comment by Blackstareye** [comment]
Is there a image which has gnome and wayland installed or is there a good guide you can recommend ?

I only find install guides with xorg and gnome.

Best case would be a build ready install image which installs gnome.

This OS seems to be quite niche

**Comment by todfox-net** [comment]
Hi! The best guide is the Void Handbook - https://docs.voidlinux.org/installation/live-images/guide.html
Once the OS is installed you can install gnome - # xbps-install -uy gnome
( https://docs.voidlinux.org/config/graphical-session/gnome.html )

Interestingly, I've noticed if I'm listening on headphones, I can toggle *off* the headset to speakers, but the reverse doesn't work. The button lights up, the indicator changes, but the audio stays on the speakers. The hotkey still does nothing, either way - the indicator doesn't change.

**Comment by Blackstareye** [comment]
I talked to some people, and I am sure that this problem is void linux specific.

I will look into it as soon I got time and check if it is easy fixable or if there is something I can do / help you with instructions. 

I will tag you / mention you here, as soon I got a solution and or more information. Okay? :)

Just FYI:
Still I will add to this issue the label "outofscope".  So that others know, that this does not affect other versions.

**Comment by todfox-net** [comment]
Thanks for the note!

By the way, I also tested this on Xorg, and the same behavior occurs, so it doesn't seem to be specific to Wayland, at least.

**Comment by Blackstareye** [comment]
Thank you for this information ! :)  
So we can circle it down to a void linux problem. Perhaps missing dependencies or something. :)

**Comment by Blackstareye** [comment]
I tried to setup everything on a vm. 

I tried gnome with wayland but get a white screen of death "oops something went wrong"

then I tried xorg and installed gnome with the help of the docu.
But I am still missing audio.

I followed the pipewire guide as well but it does not work.
Hence I can't reproduce it :/ 

This Os is quite different then any os I dealed with ^^'

I would need a fully working vm image (qemu) for that. :/ 
So I can't promise you that I find a solution for that.

I think the issue is directly related the void linux. I just cant pinpoint what goes wrong

I really tried :/
